src/smclarify/bias/report.py

```diff
@@ -547,6 +555,8 @@ def _do_report(
         return metrics_result
 
     elif facet_data_type == common.DataType.CONTINUOUS:
+        if sensitive_facet_values and len(sensitive_facet_values) > 1:
```

Comment:
Currently the len() >1 check will never be true, because if there are multiple values the data type should have been inferred as CATEGORICAL [by this hack](https://github.com/aws/amazon-sagemaker-clarify/blob/d7df2db7645f1ca0ffbde951196a219184363efa/src/smclarify/bias/metrics/common.py#L129).

The check is useful when the hack is removed. I.e. ideally we should infer the column type by the column series. I think the correct logic should be:
* series_datatype() infers the data type by the series only, i.e. remove the “values” parameter from the function
* The caller of the series_datatype() function is responsible for validation, if the label_values_or_threshold is not compatible with the inferred data type then fail the job. i.e.
  * inferred type: CONTINUOUS, label_values_or_threshold: str => failure, because the continuous column expects a numerical threshold.
  * inferred type: CONTINUOUS, label_values_or_threshold: list => failure, because the continuous column expects a single treshold.

Reply:
One of the analyzer test datasets assumes series_datatype() infers the data type when label_values_or_threshold is a list.  Removing that inference fails the test as it has 7 unique labels, but only 50 samples so it doesn't pass the uniqueness check and is marked as continuous, but then test fails since it was expecting it to be inferred as categorical.

Maybe we need to address the TODO here? https://github.com/aws/amazon-sagemaker-clarify/blob/master/src/smclarify/bias/metrics/common.py#L142-L146

One thing I can think of is if we get to the point where we are checking uniqueness factor, we can be much more lenient if the user tried to give a list of thresholds (or even 100% set it to categorical if threshold is a list)

Reply:
This was discussed pre-launch. It was decided that length of `values_or_threshold` parameter would be used as an indication from the customer on whether or not a column was a NUMERICAL or CATEGORICAL column. 

@pinaraws thoughts?

Reply:
The "hack" is doing the right thing. The `value_or_threshold` does not accept multiple intervals, so if it's a list, then the column must be categorical. This should be reevaluated if support for multiple intervals is added.